include/ghex/unstructured/grid.hpp

```diff
@@ -19,15 +19,15 @@ namespace gridtools {
 
                 template<typename Index>
                 struct grid {
-                    using index_t = Index;
+                    using index_type = Index;
                 };
 
             } // namespace detail
 
             /** @brief type to indicate unstructured grids */
             struct grid {
```

Comment:
It seems strange to have this structure here. Where is it used? Could this be simplified?

Reply:
Here my attempt is to be consistent with the type definitions used in the other pattern specialization (structured) and in the `make_pattern` function. In general, I would say that this code structure has the advantage that the user needs to specify only the grid type (choosing between the `structured` or `unstructured` struct) when calling the `make_pattern` function at the application level. The index type is then inferred directly from the domain (which needs to provide the member type `local_index_type`), and this ensures that the index type used in the Iteration space is the same used by the domain. In the unstructured case, with index type I mean the local index type.

Reply:
But why defer to a `detail_` namespace? Could this be done with just one trait class?

Reply:
Yes, I think so, but actually I do not see a way to do this without using the domain type as a template in the grid structure, which translates into an additional template parameter for the user. It could also be that I am not seeing a simpler solution, though. Anyway, modifying it will change the code in the structured specialization as well, (the same structure is used there as well).